src/client/websocket/WebSocketShard.js

```diff
@@ -1,349 +1,265 @@
 const EventEmitter = require('events');
-const { Events, OPCodes, Status, WSCodes } = require('../../util/Constants');
-const PacketManager = require('./packets/WebSocketPacketManager');
 const WebSocket = require('../../WebSocket');
+const { Status, Events, OPCodes, WSEvents, WSCodes } = require('../../util/Constants');
+let zlib;
 try {
-  var zlib = require('zlib-sync');
+  zlib = require('zlib-sync');
   if (!zlib.Inflate) zlib = require('pako');
 } catch (err) {
   zlib = require('pako');
 }
 
 /**
- * Abstracts a WebSocket connection with decoding/encoding for the Discord gateway.
- * @private
+ * Represents a Shard's Websocket connection.
  */
-class WebSocketConnection extends EventEmitter {
-  /**
-   * @param {WebSocketManager} manager The WebSocket manager
-   * @param {string} gateway The WebSocket gateway to connect to
-   */
-  constructor(manager, gateway) {
+class WebSocketShard extends EventEmitter {
```

Comment:
Is really everything supposed to be public?

Reply:
What kind of things should we make private? I am open to suggestions from everyone.